### ISSUE TITLE
Compile Error MyHwAVR.cpp:348:58: error: expected ')' before 'PRIu8'

### ISSUE THREAD

**Issue by tismoj** [mysensors/MySensors#1138]
Hi I'm new to MySensors, but I had experimented with the nRF24L01+ years ago using the maniacbug/RF24 library (though I had to modify the library to make it work).

I wanted to try MySensors, it looked straight forward enough from the videos. So I chose to try openHAB on a Raspbian, but when I was trying to compile GatewaySerial.ino on an Arduino 1.8.5 there were several compile errors, I tried installing MySensors from arduino and also tried both cloning from github and downloading MySensors-master from a link in your site (not Github) and placing it manually to where my Arduino stores its Libraries and they all have the same issue, below. Could you knidly take a look. I feel like there is just either a missing include / define, or that there are some extra include / define somewhere.    

Arduino: 1.8.5 (Mac OS X), Board: "Arduino Nano, ATmega328"

WARNING: Category '' in library EEPROM is not valid. Setting to 'Uncategorized'
WARNING: Category '' in library SPI is not valid. Setting to 'Uncategorized'
WARNING: Category '' in library SoftwareSerial is not valid. Setting to 'Uncategorized'
WARNING: Category '' in library Wire is not valid. Setting to 'Uncategorized'
WARNING: Spurious .ci folder in 'MySensors' library
WARNING: Spurious .idea folder in 'MySensors' library
WARNING: Spurious .mystools folder in 'MySensors' library
Build options changed, rebuilding all
Warning: platform.txt from core 'Arduino AVR Boards' contains deprecated recipe.ar.pattern="{compiler.path}{compiler.ar.cmd}" {compiler.ar.flags} {compiler.ar.extra_flags} "{build.path}/{archive_file}" "{object_file}", automatically converted to recipe.ar.pattern="{compiler.path}{compiler.ar.cmd}" {compiler.ar.flags} {compiler.ar.extra_flags} "{archive_file_path}" "{object_file}". Consider upgrading this core.
In file included from /Users/tismoj/Library/Arduino15/packages/arduino/hardware/avr/1.6.2/cores/arduino/Arduino.h:28:0,
                 from sketch/GatewaySerial.ino.cpp:1:
/Users/tismoj/Documents/Arduino/libraries/MySensors/hal/architecture/AVR/MyHwAVR.cpp: In function 'void hwDebugPrint(const char*, ...)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/hal/architecture/AVR/MyHwAVR.cpp:348:58: error: expected ')' before 'PRIu8'
  snprintf_P(fmtBuffer, sizeof(fmtBuffer), PSTR("0;255;%" PRIu8 ";0;%" PRIu8 ";%" PRIu32 " "),
                                                          ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyProtocolMySensors.cpp: In function 'char* protocolFormat(MyMessage&)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyProtocolMySensors.cpp:102:22: error: expected ')' before 'PRIu8'
             PSTR("%" PRIu8 ";%" PRIu8 ";%" PRIu8 ";%" PRIu8 ";%" PRIu8 ";%s\n"), message.sender,
                      ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyProtocolMySensors.cpp: In function 'char* protocolFormatMQTTTopic(const char*, MyMessage&)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyProtocolMySensors.cpp:111:25: error: expected ')' before 'PRIu8'
             PSTR("%s/%" PRIu8 "/%" PRIu8 "/%" PRIu8 "/%" PRIu8 "/%" PRIu8 ""), prefix,
                         ^
In file included from /Users/tismoj/Documents/Arduino/libraries/MySensors/MySensors.h:49:0,
                 from /Users/tismoj/Downloads/@_GitHub/MySensors/examples/GatewaySerial/GatewaySerial.ino:86:
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'void stUplinkUpdate()':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:265:44: error: expected ')' before 'PRIu8'
    TRANSPORT_DEBUG(PSTR("TSM:UPL:DGWC,O=%" PRIu8 ",N=%" PRIu8 "\n"), _transportConfig.distanceGW,
                                            ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:265:4: note: in expansion of macro 'TRANSPORT_DEBUG'
    TRANSPORT_DEBUG(PSTR("TSM:UPL:DGWC,O=%" PRIu8 ",N=%" PRIu8 "\n"), _transportConfig.distanceGW,
    ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'void stReadyTransition()':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:292:40: error: expected ')' before 'PRIu8'
  TRANSPORT_DEBUG(PSTR("TSM:READY:ID=%" PRIu8 ",PAR=%" PRIu8 ",DIS=%" PRIu8 "\n"),
                                        ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:292:2: note: in expansion of macro 'TRANSPORT_DEBUG'
  TRANSPORT_DEBUG(PSTR("TSM:READY:ID=%" PRIu8 ",PAR=%" PRIu8 ",DIS=%" PRIu8 "\n"),
  ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'void stFailureTransition()':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:342:40: error: expected ')' before 'PRIu8'
  TRANSPORT_DEBUG(PSTR("TSM:FAIL:CNT=%" PRIu8 "\n"),_transportSM.failureCounter);
                                        ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:342:2: note: in expansion of macro 'TRANSPORT_DEBUG'
  TRANSPORT_DEBUG(PSTR("TSM:FAIL:CNT=%" PRIu8 "\n"),_transportSM.failureCounter);
  ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'bool transportWaitUntilReady(uint32_t)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:449:38: error: expected ')' before 'PRIu32'
  TRANSPORT_DEBUG(PSTR("TSF:WUR:MS=%" PRIu32 "\n"), waitingMS); // timeout
                                      ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:449:2: note: in expansion of macro 'TRANSPORT_DEBUG'
  TRANSPORT_DEBUG(PSTR("TSF:WUR:MS=%" PRIu32 "\n"), waitingMS); // timeout
  ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'bool transportCheckUplink(bool)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:485:44: error: expected ')' before 'PRIu8'
    TRANSPORT_DEBUG(PSTR("TSF:CKU:DGWC,O=%" PRIu8 ",N=%" PRIu8 "\n"), _transportConfig.distanceGW,
                                            ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:485:4: note: in expansion of macro 'TRANSPORT_DEBUG'
    TRANSPORT_DEBUG(PSTR("TSF:CKU:DGWC,O=%" PRIu8 ",N=%" PRIu8 "\n"), _transportConfig.distanceGW,
    ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'bool transportAssignNodeID(uint8_t)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:504:42: error: expected ')' before 'PRIu8'
   TRANSPORT_DEBUG(PSTR("TSF:SID:OK,ID=%" PRIu8 "\n"),newNodeId); // Node ID assigned
                                          ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:504:3: note: in expansion of macro 'TRANSPORT_DEBUG'
   TRANSPORT_DEBUG(PSTR("TSF:SID:OK,ID=%" PRIu8 "\n"),newNodeId); // Node ID assigned
   ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:507:45: error: expected ')' before 'PRIu8'
   TRANSPORT_DEBUG(PSTR("!TSF:SID:FAIL,ID=%" PRIu8 "\n"),newNodeId); // ID is invalid, cannot assign ID
                                             ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:507:3: note: in expansion of macro 'TRANSPORT_DEBUG'
   TRANSPORT_DEBUG(PSTR("!TSF:SID:FAIL,ID=%" PRIu8 "\n"),newNodeId); // ID is invalid, cannot assign ID
   ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'bool transportRouteMessage(MyMessage&)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:534:38: error: expected ')' before 'PRIu8'
    TRANSPORT_DEBUG(PSTR("!TSF:RTE:%" PRIu8 " UNKNOWN\n"), destination); // route unknown
                                      ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:534:4: note: in expansion of macro 'TRANSPORT_DEBUG'
    TRANSPORT_DEBUG(PSTR("!TSF:RTE:%" PRIu8 " UNKNOWN\n"), destination); // route unknown
    ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'uint8_t transportPingNode(uint8_t)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:612:44: error: expected ')' before 'PRIu8'
   TRANSPORT_DEBUG(PSTR("TSF:PNG:SEND,TO=%" PRIu8 "\n"), targetId);
                                            ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:612:3: note: in expansion of macro 'TRANSPORT_DEBUG'
   TRANSPORT_DEBUG(PSTR("TSF:PNG:SEND,TO=%" PRIu8 "\n"), targetId);
   ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'void transportProcessMessage()':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:660:40: error: expected ')' before 'PRIu8'
  TRANSPORT_DEBUG(PSTR("TSF:MSG:READ,%" PRIu8 "-%" PRIu8 "-%" PRIu8 ",s=%" PRIu8 ",c=%" PRIu8 ",t=%"
                                        ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:660:2: note: in expansion of macro 'TRANSPORT_DEBUG'
  TRANSPORT_DEBUG(PSTR("TSF:MSG:READ,%" PRIu8 "-%" PRIu8 "-%" PRIu8 ",s=%" PRIu8 ",c=%" PRIu8 ",t=%"
  ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:669:41: error: expected ')' before 'PRIu8'
   TRANSPORT_DEBUG(PSTR("!TSF:MSG:LEN,%" PRIu8 "!=%" PRIu8 "\n"), payloadLength,
                                         ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:669:3: note: in expansion of macro 'TRANSPORT_DEBUG'
   TRANSPORT_DEBUG(PSTR("!TSF:MSG:LEN,%" PRIu8 "!=%" PRIu8 "\n"), payloadLength,
   ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:677:42: error: expected ')' before 'PRIu8'
   TRANSPORT_DEBUG(PSTR("!TSF:MSG:PVER,%" PRIu8 "!=%" PRIu8 "\n"), mGetVersion(_msg),
                                          ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:677:3: note: in expansion of macro 'TRANSPORT_DEBUG'
   TRANSPORT_DEBUG(PSTR("!TSF:MSG:PVER,%" PRIu8 "!=%" PRIu8 "\n"), mGetVersion(_msg),
   ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:775:49: error: expected ')' before 'PRIu8'
      TRANSPORT_DEBUG(PSTR("TSF:MSG:PINGED,ID=%" PRIu8 ",HP=%" PRIu8 "\n"), sender,
                                                 ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:775:6: note: in expansion of macro 'TRANSPORT_DEBUG'
      TRANSPORT_DEBUG(PSTR("TSF:MSG:PINGED,ID=%" PRIu8 ",HP=%" PRIu8 "\n"), sender,
      ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:789:53: error: expected ')' before 'PRIu8'
       TRANSPORT_DEBUG(PSTR("TSF:MSG:PONG RECV,HP=%" PRIu8 "\n"),
                                                     ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:789:7: note: in expansion of macro 'TRANSPORT_DEBUG'
       TRANSPORT_DEBUG(PSTR("TSF:MSG:PONG RECV,HP=%" PRIu8 "\n"),
       ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:856:52: error: expected ')' before 'PRIu8'
       TRANSPORT_DEBUG(PSTR("TSF:MSG:FPAR REQ,ID=%" PRIu8 "\n"), sender); // FPAR: find parent request
                                                    ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:856:7: note: in expansion of macro 'TRANSPORT_DEBUG'
       TRANSPORT_DEBUG(PSTR("TSF:MSG:FPAR REQ,ID=%" PRIu8 "\n"), sender); // FPAR: find parent request
       ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:925:52: error: expected ')' before 'PRIu8'
       TRANSPORT_DEBUG(PSTR("TSF:MSG:REL PxNG,HP=%" PRIu8 "\n"), hopsCnt);
                                                    ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:925:7: note: in expansion of macro 'TRANSPORT_DEBUG'
       TRANSPORT_DEBUG(PSTR("TSF:MSG:REL PxNG,HP=%" PRIu8 "\n"), hopsCnt);
       ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'bool transportSendWrite(uint8_t, MyMessage&)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:1001:42: error: expected ')' before 'PRIu8'
  TRANSPORT_DEBUG(PSTR("%sTSF:MSG:SEND,%" PRIu8 "-%" PRIu8 "-%" PRIu8 "-%" PRIu8 ",s=%" PRIu8 ",c=%"
                                          ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:1001:2: note: in expansion of macro 'TRANSPORT_DEBUG'
  TRANSPORT_DEBUG(PSTR("%sTSF:MSG:SEND,%" PRIu8 "-%" PRIu8 "-%" PRIu8 "-%" PRIu8 ",s=%" PRIu8 ",c=%"
  ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'void transportReportRoutingTable()':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:1086:45: error: expected ')' before 'PRIu8'
    TRANSPORT_DEBUG(PSTR("TSF:RRT:ROUTE N=%" PRIu8 ",R=%" PRIu8 "\n"), cnt, route);
                                             ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:1086:4: note: in expansion of macro 'TRANSPORT_DEBUG'
    TRANSPORT_DEBUG(PSTR("TSF:RRT:ROUTE N=%" PRIu8 ",R=%" PRIu8 "\n"), cnt, route);
    ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp: In function 'int16_t transportSignalReport(char)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:1174:39: error: expected ')' before 'PRIu8'
  TRANSPORT_DEBUG(PSTR("TSF:SIR:CMD=%" PRIu8 ",VAL=%" PRIu16 "\n"), reportCommand, result);
                                       ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MyTransport.cpp:1174:2: note: in expansion of macro 'TRANSPORT_DEBUG'
  TRANSPORT_DEBUG(PSTR("TSF:SIR:CMD=%" PRIu8 ",VAL=%" PRIu16 "\n"), reportCommand, result);
  ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MySensorsCore.cpp: In function 'void _begin()':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MySensorsCore.cpp:180:42: error: expected ')' before 'PRIu8'
  CORE_DEBUG(PSTR("MCO:BGN:INIT OK,TSP=%" PRIu8 "\n"), isTransportReady() && transportSanityCheck());
                                          ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MySensorsCore.cpp:180:2: note: in expansion of macro 'CORE_DEBUG'
  CORE_DEBUG(PSTR("MCO:BGN:INIT OK,TSP=%" PRIu8 "\n"), isTransportReady() && transportSanityCheck());
  ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MySensorsCore.cpp: In function 'int8_t _sleep(uint32_t, bool, uint8_t, uint8_t, uint8_t, uint8_t)':
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MySensorsCore.cpp:551:33: error: expected ')' before 'PRIu32'
  CORE_DEBUG(PSTR("MCO:SLP:MS=%" PRIu32 ",SMS=%" PRIu8 ",I1=%" PRIu8 ",M1=%" PRIu8 ",I2=%" PRIu8
                                 ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/MyConfig.h:2077:43: note: in definition of macro 'DEBUG_OUTPUT'
 #define DEBUG_OUTPUT(x,...)  hwDebugPrint(x, ##__VA_ARGS__) //!< debug
                                           ^
/Users/tismoj/Documents/Arduino/libraries/MySensors/core/MySensorsCore.cpp:551:2: note: in expansion of macro 'CORE_DEBUG'
  CORE_DEBUG(PSTR("MCO:SLP:MS=%" PRIu32 ",SMS=%" PRIu8 ",I1=%" PRIu8 ",M1=%" PRIu8 ",I2=%" PRIu8
  ^
exit status 1
Error compiling for board Arduino Nano.

This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

**Comment by per1234** [comment]
You're using a very outdated version of Arduino AVR Boards, which is causing these errors. To update, do this:

1. Tools > Board > Boards Manager
2. Wait for downloads to finish.
3. Click on "Arduino AVR Boards".
4. Click "Update".
5. Wait for update to finish.
6. Click "Close".

**Comment by tismoj** [comment]
Thanks finally that solve my issue. and I was able to upload GatewaySerial.ino. Though, I've noticed that the setup and loop functions doesn't have anything on it. I doubt that it is gonna run from just defines and includes. And How do I know that it is working? from the console?

Note, I am currently at https://github.com/tobof/openhab2-addons/wiki/Configuration, I have already installed the openHAB Add-on MySensors.
